src/main/java/com/microsoft/sqlserver/jdbc/TimeoutPoller.java

```diff
@@ -0,0 +1,75 @@
+package com.microsoft.sqlserver.jdbc;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+
+/**
+ * Thread that runs in the background while the mssql driver is used that can timeout TDSCommands Checks all registered
+ * commands every second to see if they can be interrupted
+ */
+public final class TimeoutPoller implements Runnable {
+    private List<TimeoutCommand> timeoutCommands = new ArrayList<>();

```

Comment:
Please parametrize `TimeoutCommand`, convert references `TimeoutCommand` to `TimeoutCommand<TDSCommand>` to remove the warnings.

Reply:
done